src/main/java/ch/uzh/ifi/hase/soprafs22/rest/mapper/DTOMapper.java

```diff
@@ -36,6 +36,23 @@ public interface DTOMapper {
   @Mapping(source = "gender", target = "gender")
   UserGetDTO convertEntityToUserGetDTO(User user);
 
+
+  @Mapping(source = "minAge", target = "minAge")
+  @Mapping(source = "maxAge", target = "maxAge")
+  @Mapping(source = "genderPreferences", target = "genderPreferences")
+  UserGetPreferencesDTO convertEntityToUserPreferencesGetDTO(User user);
+
+  @Mapping(source = "id", target = "id")
+  @Mapping(source = "minAge", target = "minAge")
+  @Mapping(source = "maxAge", target = "maxAge")
+  @Mapping(source = "genderPreferences", target = "genderPreferences")
+  User convertUserPreferencesPutDTOtoEntity(UserPreferencesPutDTO userPutPreferencesDTO);
+
+  @Mapping(source = "id", target = "id")
+  @Mapping(source = "username", target = "username")
+  @Mapping(source = "birthday", target = "birthday")
+  User convertUserPutDTOtoEntity(UserPutDTO userPutDTO);
```

Comment:
I would merge that ( = make one (bigger) UserPutDTO), and make a general updateUser method, which updates all fields that are given in the DTO. Also, does the frontend give the userId in the body? I think not (only as a path variable). So remove the "id" mapping (also in the DTO). Furthermore, since we don't allow for changing birthday, you can remove that as well.

Reply:
Done